pkg/provider/pulumi/generator.go

```diff
@@ -75,11 +75,17 @@ func New(s *project.Project, t *stack.Config) (types.Provider, error) {
 		return nil, err
 	}
 	*/
-	return &pulumiDeployment{
+	pDep := &pulumiDeployment{
 		s: s,
 		t: t,
 		p: prov,
-	}, nil
+	}
+
+	if err := pDep.p.Validate(); err != nil {
```

Comment:
this is been done now first thing in the cmd.Up

Reply:
commit dropped from rabase